### PR TITLE
Melee Weapon

### PR DESCRIPTION
**Pull request by Zac8668**
Add Melee Weapon
#86

### CODE REVIEW

**Review by zicklag — APPROVED** [review]
This looks great!

The only thing I notice is that we might want to tweak a weapon offset of some sort for the positioning relative to the player.

image

Still, I don't think that needs to block this PR, and we can do that after this is merged.

**Comment by Zac8668** [comment]
> The only thing I notice is that we might want to tweak a weapon offset of some sort for the positioning relative to the player.
> 
> image

@zicklag maybe this could be a field on the yaml file? Cause there's a variety of weapons that could be implemented, and with that we could have a negative offset, something that a player relative offset would not probably not allow, and as I noticed it now, the sword on jumpy has a negative one: 
image

**Comment by zicklag** [comment]
Yeah, that's exactly what I was thinking. :+1:

File: src/metadata.rs
Comment (odecay) [inline src/metadata.rs:142]: Is there a reason we need to box here? (not saying its wrong, im just not super acquainted with Boxing)

File: src/player.rs
Comment (odecay) [inline src/player.rs:76]: :+1:

**Review by odecay — COMMENTED** [review]
Looks good.

File: src/metadata.rs
Comment (Zac8668) [reply src/metadata.rs:142]: Clippy said it's best because of the size difference between the ItemKind Variants, I was surprised too

Comment (zicklag) [reply src/metadata.rs:142]: Yeah, because an enum must take up as much memory as it's largest variant, and the `FighterSpritesheetMeta` struct makes the `MeleeWeapon` variant much larger than all the other variants, boxing it can make the enum take less space when using the non-`MeleeWeapon` variants.

I'm not sure if it's really a concern, but seems like we might as well trust clippy in this case.

**Comment by zicklag** [comment]
Awesome, thanks a bunch!

bors merge

**Pull request merged**
